### PR TITLE
sci-misc/boinc-7.6.25.ebuild, sci-misc/boinc-7.6.31.ebuild

### PR DESCRIPTION
**Pull request by Yamakuzure**
Minor version bumps.

No changes from boinc-7.4.42-r1.ebuild but the version numbers.

@jlec @SoapGentoo : Justin, David, I hope I did it right on my first attempt. :-)

### CODE REVIEW

File: sci-misc/boinc/boinc-7.6.25.ebuild [hunk 1]
```diff
@@ -0,0 +1,141 @@
+# Copyright 1999-2015 Gentoo Foundation
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: Year wrong

File: sci-misc/boinc/boinc-7.6.25.ebuild [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=5
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: We prefer to only have `EAPI=6` for new code.

File: sci-misc/boinc/boinc-7.6.25.ebuild [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+#WANT_AUTOMAKE="1.11"
+
+AUTOTOOLS_AUTORECONF=true
+
+inherit autotools-utils eutils flag-o-matic systemd user versionator wxwidgets
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: with `EAPI=6`, `autotools-utils.eclass` was deprecated. Just switch back to calling `eautoreconf` in `src_prepare`

File: sci-misc/boinc/boinc-7.6.25.ebuild [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+
+S="${WORKDIR}/${PN}-client_release-${MY_PV}-${PV}"
+
+AUTOTOOLS_IN_SOURCE_BUILD=1
+
+src_prepare() {
+	# prevent bad changes in compile flags, bug 286701
+	sed -i -e "s:BOINC_SET_COMPILE_FLAGS::" configure.ac || die "sed failed"
+
+	autotools-utils_src_prepare
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: Once you switch to `EAPI=6`, you can replace all calls to the parent phase functions with `default`.

File: sci-misc/boinc/boinc-7.6.25.ebuild [hunk 5]
```diff
@@ -0,0 +1,141 @@
...
+}
+
+src_configure() {
+	local myeconfargs=(
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: get rid of the array, attach everything to `econf` (don't forget to escape the newlines!)

File: sci-misc/boinc/boinc-7.6.25.ebuild [hunk 6]
```diff
@@ -0,0 +1,141 @@
...
+	)
+
+	# look for wxGTK
+	if use X; then
+		WX_GTK_VER="3.0"
+		need-wxwidgets unicode
+		myeconfargs+=(--with-wx-config="${WX_CONFIG}")
+	else
+		myeconfargs+=(--without-wxdir)
+	fi
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: parts of the conditional can be attached to `econf`, with `$(usex X --with-wx-config="${WX_CONFIG}" --without-wxdir)`

File: sci-misc/boinc/boinc-7.6.25.ebuild [hunk 7]
```diff
@@ -0,0 +1,141 @@
...
+
+	keepdir /var/lib/${PN}
+
+	if use X; then
+		newicon "${S}"/packages/generic/sea/${PN}mgr.48x48.png ${PN}.png
+		make_desktop_entry boincmgr "${PN}" "${PN}" "Math;Science" "Path=/var/lib/${PN}"
+	fi
+
+	# cleanup cruft
+	rm -rf "${ED}"/etc
```
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: missing `|| die`

**Comment by SoapGentoo** [comment]
Dear Sven,
thanks for the PR. I've commented on some issues (that aren't your fault, but while we're at it, let's fix the old stuff). Also, your commit message is still off. It needs to be of the form `CATEGORY/PN:`. Also, please split up the single commit into two, one for each ebuild (in order to make reverts easier). You can get the formatting right if you consistently use repoman, which handles these things better than `git commit`. Also please include the Gentoo bug number below, so we have something to reference. You should get something along the lines of:
```
sci-visualization/ggobi: Fix -Werror=format-security errors

Gentoo-Bug: 577430

Package-Manager: portage-2.2.28
```
The last line is not that important, it comes from repoman (which I nevertheless strongly recommend).

**Comment by Yamakuzure** [comment]
Dear David,

thank you very much for your review and your suggestions. I am learning a lot!

The ebuilds are based on sci-misc/boinc-7.4.42-r1. They aren't doing anything new, it is just the version number that does it all.

I wanted to clean up the old ebuilds, too, once I got my first PR through.

I will fix the issues you found and ask repoman for any other mistakes like you suggested.

Thanks again!

File: sci-misc/boinc/boinc-7.6.25.ebuild (hunk 2 above)
Comment (Yamakuzure) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: Am Dienstag, 15. März 2016, 06:11:57 CET schrieb David Seifert:
> > +EAPI=5
> 
> We prefer to only have `EAPI=6` for new code.

Unfortunately wxwidgets.eclass dies then, so I have to stay with EAPI=5 for now until the eclass is upgraded.

 * ERROR: sci-misc/boinc-7.6.31::gentoo failed (depend phase):
 *   EAPI=6 is not supported
 * 
 * Call stack:
 *             ebuild.sh, line 624:  Called source '/home/sed/pryde/PrydeWorX/gentoo_myfork/sci-misc/boinc/boinc-7.6.31.ebuild'
 *   boinc-7.6.31.ebuild, line   7:  Called inherit 'eutils' 'flag-o-matic' 'systemd' 'user' 'versionator' 'wxwidgets'
 *             ebuild.sh, line 320:  Called __qa_source '/home/sed/pryde/PrydeWorX/gentoo_myfork/eclass/wxwidgets.eclass'
 *             ebuild.sh, line 120:  Called source '/home/sed/pryde/PrydeWorX/gentoo_myfork/eclass/wxwidgets.eclass'
 *      wxwidgets.eclass, line  68:  Called die
 * The specific snippet of code:
 *              die "EAPI=${EAPI:-0} is not supported"

**Comment by Yamakuzure** [comment]
I am closing this pull request and fix the issues first.

File: sci-misc/boinc/boinc-7.6.25.ebuild (hunk 2 above)
Comment (SoapGentoo) [inline sci-misc/boinc/boinc-7.6.25.ebuild]: Dear Sven,
I totally overlooked `wxwidgets.eclass`. So yes, stay at `EAPI=5` for the moment, but make it `EAPI=6`-ready, i.e., remove `autotools-utils`, and include the rest of the improvements.

**Pull request closed without merging**